QPM/SymLinker/Linker.cs

```diff
@@ -0,0 +1,89 @@
+﻿using SymLinker.LinkCreators;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+
+// Stolen from https://github.com/huffSamuel/SymLinker
+// Cleaned up by Fern, because the old code was horrible
+namespace SymLinker
+{
+    public class Linker
+    {
+        private readonly ISymLinkCreator? _linker;
+
+        /// <summary>
+        /// Creates a new Symbolic Link Creator
+        /// </summary>
+        public Linker()
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                _linker = new WindowsSymLinkCreator();
+            }
+            // TODO: Implement
+            // else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            // {
+            //     _linker = new LinuxSymLinkCreator();
+            // }
+            // else
+            // {
+            //     _linker = new OSXSymLinkCreator();
+            // }
+        }
+
+        public bool IsValid() => _linker != null;
+
+        /// <summary>
+        /// Creates a symbolic link from <paramref name="source"/> to <paramref name="dest"/>
+        /// </summary>
+        /// <param name="source">Source file</param>
+        /// <param name="dest">Destination directory</param>
+        /// <returns>
+        /// Returns true if the system was able to create the SymLink
+        /// </returns>
+        public string? CreateLink(string source, string dest)
+        {
+            if (!IsValid())
+            {
+                return "Platform does not support symlinking or hard linking yet";
+            }
+
+            var error = CheckLinkReadiness(source, dest);
+
+            if (error != null)
+            {
+                return error;
+            }
+
+            var linkMade = _linker.CreateSymLink(source, dest, true);
```

Comment:
Either _explicitly_ null check `_linker` before this call or do `_linker!.CreateSymLink`

Reply:
If you notice, the `IsValid();` takes care of the null check so I believe it is unnecessary though I'll do it anyways

Reply:
There's no guarantee that `IsValid` will _always_ be performing a null check, so you could make it caller responsibility to call `IsValid` beforehand and have a call portion for null check, or you could just cut some corners and use `!`. Either way, I want that warning gone :)

(wanna have good style for incoming QPM things even if everything else in QPM is a mess)